### DISCUSSION TITLE
Why is it not possible for sform_code to be set to 1 through the set_sform() function in nifti1.py?

### DISCUSSION THREAD

**Discussion by hbhutta** [nipy/nibabel#1395] [Q&A]
From the function `nifti1.set_sform()`:

```
* If affine is None, `code`-> 0
* If affine not None and existing sform code in header == 0, `code`-> 2 (aligned)
* If affine not None and existing sform code in header != 0, `code`-> existing sform code in header
```

So if the original NIFTI file has an sform_code of 0, and a null sform affine, and I try to set the sform to something that is not None, it just sets the sform code to 2. I am actually trying to set the sform_code *and* change the sform matrix for this NIFTI file using the setters in nifti1.py but no change takes place.

**Comment by effigies** [comment]
`img.header.set_sform(affine, code)` should work. What did you try, and what happened?
